sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaSinkMetrics.java

```diff
@@ -97,7 +86,15 @@ public static Gauge createBacklogGauge(String topic, int partitionId) {
    * @return Counter.
    */
   public static Gauge createBacklogGauge(MetricName name) {
-    return new DelegatingGauge(name, false, true);
+    // Use label to differenciate between the type of gauge metric is created
+    // TODO(34195):
+    if (name.getLabels().containsKey(MonitoringInfoConstants.Labels.PER_WORKER_METRIC)
+        && name.getLabels().get(MonitoringInfoConstants.Labels.PER_WORKER_METRIC).equals("true")) {
+      return new DelegatingGauge(name, false);
+    } else {
+      // Currently KafkaSink metrics only supports aggregated per worker metrics.
```

Comment:
should we make it a Preconditions.checkState(isPerWorkerMetric))?

seems perhaps better to fail so that if more metrics are added it's  not confusing why they don't do anything